statsmodels/nonparametric/kde.py

```diff
@@ -454,7 +457,10 @@ def kdensityfft(X, kernel="gau", bw="normal_reference", weights=None, gridsize=N
     kern = kernel_switch[kernel]()
 
     try:
-        bw = float(bw)
+        if callable(bw):
```

Comment:
The if should be before the try. If the callable fails, we don't want to hit the except.